src/compile/selection/assemble.ts

```diff
@@ -113,9 +123,22 @@ export function assembleTopLevelSignals(model: UnitModel, signals: any[]) {
 
 export function assembleUnitSelectionData(model: UnitModel, data: VgData[]): VgData[] {
   forEachSelection(model, selCmpt => {
+    const dataObj = {name: selCmpt.name + STORE};
+    if (selCmpt.init) {
+      const fields = selCmpt.project.items.map(proj => {
+        const {signals, ...rest} = proj;
+        return rest;
+      });
+      const insert = selCmpt.init.map((i: SelectionInit | SelectionInit[]) => assembleInitData(i));
+      if (selCmpt.type === 'interval') {
+        dataObj['values'] = {unit: unitName(model), fields, values: insert};
```

Comment:
Flagging this based on our in-person conversation: the square brackets notation circumvents TypeScript, so we want to use it as an absolute last resort. In this case, start by annotating `dataObj` (perhaps better called an `init`?) to be a `VgData` and then assign an array to `values`.

An additional stylistic point here -- we tend to prefer ternary assignments for simple if-then-else.